cache/lib/cache/s3_upload_worker.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Cache.S3UploadWorker do
+  @moduledoc """
+  Oban worker for uploading CAS artifacts to S3.
+  """
+
+  use Oban.Worker, queue: :s3_uploads, max_attempts: 3
+
+  require Logger
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"key" => key}}) do
+    Logger.info("Starting S3 upload for artifact: #{key}")
+
+    local_path = Cache.Disk.artifact_path(key)
+
+    if File.exists?(local_path) do
+      case upload_to_s3(key, local_path) do
+        :ok ->
+          Logger.info("Successfully uploaded artifact to S3: #{key}")
+          :ok
+
+        {:error, reason} ->
+          Logger.error("Failed to upload artifact to S3 #{key}: #{inspect(reason)}")
+          {:error, reason}
+      end
+    else
+      Logger.warning("Skipping S3 upload for #{key}: file not found at #{local_path}")
+      {:cancel, :file_not_found}
+    end
+  end
+
+  def enqueue_upload(account_handle, project_handle, id) do
+    key = Cache.Disk.cas_key(account_handle, project_handle, id)
+
+    %{key: key}
+    |> __MODULE__.new()
+    |> Oban.insert()
+  end
+
+  defp upload_to_s3(key, local_path) do
+    bucket = Application.get_env(:cache, :s3)[:bucket]
```

Comment:
What about this to fail right away instead of ending with a `nil` further down if the bucket is missing?

```suggestion
    bucket = Application.get_env(:cache, :s3) |> Keyword.fetch!(:bucket)
```